### DISCUSSION TITLE
How to force WinUI app to auto update on startup (WAP) Question: [your question here]

### DISCUSSION THREAD

**Discussion by EugenM88** [microsoft/microsoft-ui-xaml#7825] [General]
Greetings, i've been using the publish feature of the WAP WinUI3 Desktop for a while now but for some reason i can never get ito to update automaticaly on startup, i even tried increasing the Assembly version hoping that would fix the problem however not even that made the app to check for any update whatsoever.

Maybe im missing something, can anyone please tell me what i must do for this app to update automaticaly on startup?

Thanks in advance!

**Comment by dongle-the-gadget** [comment]
You would need to combine WAP with some method of auto updating, such as `.appinstaller` or the Microsoft Store.

To check for updates, you can use these APIs:
**AppInstaller:** https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.package.checkupdateavailabilityasync
**Store:** https://docs.microsoft.com/en-us/windows/uwp/packaging/self-install-package-updates

**Comment by EugenM88** [comment]
Greetings.

Thank you very much for your reply.
Im sorry but im a little confused, why would i need to implement my own updating process if wap package already has one implemented?
When creating the installer, visual studio asks for location of the updated app version.

My only problem is that i have no idea how to trigger it.

Shouldn't it be an automatic process just like a clickonce deployment?

**Comment by dongle-the-gadget** [comment]
Microsoft Store and `.appinstaller` both provides auto updating. The options above are for when you need to manually check for updates.

The option to automatically update a sideloaded app in Visual Studio actually generates an `.appinstaller` file.

**Comment by EugenM88** [comment]
Greetings!
Thank you for your help.
I think i finally figured out how this update system works...

It seems the updating process is very different from the clickonce system.

For once you must install the app from the root directory or else the auto update process wont work.

The updating process runs silently as soon as you execute the app, no progress or notification is ever shown to the user that an update is underway.

Finally once you close the app after a few moments and start it again the app is now updated to the latest version.
